src/App.tsx

```diff
@@ -15,48 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+type StateVar = {
+  sortBy: string,
+  isReverse: boolean,
+};
+
+type Func = (goods: string[], { sortBy, isReverse }: StateVar) => string[];
+
+const getPrepareGoods: Func = (goods, { sortBy, isReverse }) => {
+  const prepareGoods = [...goods];
+
+  if (sortBy) {
+    prepareGoods.sort((goodA, goodB) => {
+      switch (sortBy) {
+        case SortBy.Alphabet:
+          return goodA.localeCompare(goodB);
+        case SortBy.Length:
+          return goodA.length - goodB.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+
+  const goods = getPrepareGoods(goodsFromServer, { sortBy, isReverse });
+
+  const GoodItems = () => (
+    <ul>
+      {goods.map(good => (
+        <li key={good} data-cy="Good">{good}</li>
+      ))}
+    </ul>
+  );
```

Comment:
Think you can render ul without creating func ( component*) ( not critical, recommendation)